src/default-options-handler.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GitHubPr, GitHubPrUserOptions} from './types';
+
+const defaultNameNoSpaces = 'code-suggestions';
+const defaultDescription = 'code suggestions';
+const defaultMessage = 'chore: code suggestions';
+const defaultPrimaryBranch = 'master';
+
+/**
+ * Add defaults to GitHub PR data
+ * @param {GitHubPrUserOptions} options the user-provided github context
+ * @returns {GitHubPr} git hub context with defaults applied
+ */
+function addPrOptionDefaults(options: GitHubPrUserOptions): GitHubPr {
+  const gitHubPr: GitHubPr = {
+    upstreamOwner: options.upstreamOwner,
+    upstreamRepo: options.upstreamRepo,
+    branch: options.branch || defaultNameNoSpaces,
+    description: options.description || defaultDescription,
+    title: options.title || defaultMessage,
+    force: options.force || false,
+    message: options.message || defaultMessage,
+    primary: options.primary || defaultPrimaryBranch,
+    maintainersCanModify: options.maintainersCanModify === false ? false : true,
+  };
+  return gitHubPr;
```

Comment:
Should the core library be dictating the default values or should the invoker (CLI, GitHub action)?

Reply:
Good question. My answer to that in short is yes and no. In my experience when using libraries there are often defaults so that the user does not need to set values for every single option. So yes some behaviour is natural to stay in the framework-core. Maybe the strings are less intuitive so that can be moved to the CLI. But this seems more grey-area to me. What do you think?

Reply:
Also it's worth mentioning that we have some default file mode established in `FileData`